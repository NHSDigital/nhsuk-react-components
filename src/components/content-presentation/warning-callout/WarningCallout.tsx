import React, { FC, HTMLProps } from 'react';
import classNames from 'classnames';
import HeadingLevel, { HeadingLevelType } from '@util/HeadingLevel';

interface WarningCalloutLabelProps extends HTMLProps<HTMLHeadingElement> {
  headingLevel?: HeadingLevelType;
  visuallyHiddenText?: string | false;
}

const WarningCalloutLabel: FC<WarningCalloutLabelProps> = ({
  className,
  visuallyHiddenText = 'Important: ',
  children,
  ...rest
}) => (
  <HeadingLevel className={classNames('nhsuk-warning-callout__label', className)} {...rest}>
    {/* eslint-disable-next-line jsx-a11y/aria-role */}
    <span role="text">
      {visuallyHiddenText && <span className="nhsuk-u-visually-hidden">{visuallyHiddenText}</span>}
      {children}
    </span>
  </HeadingLevel>
);

interface IWarningCallout extends FC<HTMLProps<HTMLDivElement>> {
  Label: typeof WarningCalloutLabel;
}

const WarningCallout: IWarningCallout = ({ className, ...rest }) => (
  <div className={classNames('nhsuk-warning-callout', className)} {...rest} />
);

WarningCallout.Label = WarningCalloutLabel;

export default WarningCallout;
