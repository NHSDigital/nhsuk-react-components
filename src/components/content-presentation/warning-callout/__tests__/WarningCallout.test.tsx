import React from 'react';
import { render } from '@testing-library/react';
import WarningCallout from '../WarningCallout';

describe('WarningCallout', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <WarningCallout>
        <WarningCallout.Label>School, nursery or work</WarningCallout.Label>
        <p>
          Stay away from school, nursery or work until all the spots have crusted over. This is
          usually 5 days after the spots first appeared.
        </p>
      </WarningCallout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('adds default visually hidden text', () => {
    const { container } = render(
      <WarningCallout>
        <WarningCallout.Label>School, nursery or work</WarningCallout.Label>
        <p>
          Stay away from school, nursery or work until all the spots have crusted over. This is
          usually 5 days after the spots first appeared.
        </p>
      </WarningCallout>,
    );

    expect(container.querySelector('.nhsuk-warning-callout__label')?.textContent).toBe(
      'Important: School, nursery or work',
    );
  });

  it('adds custom visually hidden text', () => {
    const { container } = render(
      <WarningCallout>
        <WarningCallout.Label visuallyHiddenText="Not Very Important: ">
          School, nursery or work
        </WarningCallout.Label>
        <p>
          Stay away from school, nursery or work until all the spots have crusted over. This is
          usually 5 days after the spots first appeared.
        </p>
      </WarningCallout>,
    );

    expect(container.querySelector('.nhsuk-warning-callout__label')?.textContent).toBe(
      'Not Very Important: School, nursery or work',
    );
  });

  it('can disable visually hidden text', () => {
    const { container } = render(
      <WarningCallout>
        <WarningCallout.Label visuallyHiddenText={false}>
          School, nursery or work
        </WarningCallout.Label>
        <p>
          Stay away from school, nursery or work until all the spots have crusted over. This is
          usually 5 days after the spots first appeared.
        </p>
      </WarningCallout>,
    );

    expect(container.querySelector('.nhsuk-warning-callout__label')?.textContent).toBe(
      'School, nursery or work',
    );
  });
});
