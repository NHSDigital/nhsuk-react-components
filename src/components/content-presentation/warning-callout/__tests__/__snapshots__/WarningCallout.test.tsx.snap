// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningCallout matches snapshot 1`] = `
<div>
  <div
    class="nhsuk-warning-callout"
  >
    <h3
      class="nhsuk-warning-callout__label"
    >
      <span
        role="text"
      >
        <span
          class="nhsuk-u-visually-hidden"
        >
          Important: 
        </span>
        School, nursery or work
      </span>
    </h3>
    <p>
      Stay away from school, nursery or work until all the spots have crusted over. This is usually 5 days after the spots first appeared.
    </p>
  </div>
</div>
`;
